### ISSUE TITLE
[v0.1.6] Thread dies immediately.

### ISSUE THREAD

**Issue by nownabe** [meh/ruby-thread#34]
Hi, I was faced with a problem using v0.1.6.
With 0.1.6, threads created by `Thread.pool` dies immediately.

Follwing is with v0.1.6, worker is `dead`.

```
[1] pry(main)> require "thread/pool"
=> true
[2] pry(main)> Thread.pool(1)
=> #<Thread::Pool:0x007fdb3d2957f0
 @auto_trim=false,
 @block=nil,
 @cond=#<Thread::ConditionVariable:0x007fdb3d2957c8>,
 @done=#<Thread::ConditionVariable:0x007fdb3d295750>,
 @done_mutex=#<Mutex:0x007fdb3d295700>,
 @idle_trim=nil,
 @max=1,
 @min=1,
 @mutex=#<Mutex:0x007fdb3d295778>,
 @shutdown=false,
 @spawned=1,
 @timeouts={},
 @todo=[],
 @trim_requests=0,
 @waiting=1,
 @workers=[#<Thread:0x007fdb3d295660@/Users/nownabe/tmp/threadpool/vendor/bundle/ruby/2.2.0/gems/thread-0.1.6/lib/thread/pool.rb:363 dead>]>
```

Following is with v0.1.5, worker is `sleep`.

```
[1] pry(main)> require "thread/pool"
=> true
[2] pry(main)> Thread.pool(1)
=> #<Thread::Pool:0x007fbc530f1fd8
 @auto_trim=false,
 @block=nil,
 @cond=#<Thread::ConditionVariable:0x007fbc530f1fb0>,
 @done=#<Thread::ConditionVariable:0x007fbc530f1f38>,
 @done_mutex=#<Mutex:0x007fbc530f1ee8>,
 @idle_trim=nil,
 @max=1,
 @min=1,
 @mutex=#<Mutex:0x007fbc530f1f60>,
 @shutdown=false,
 @spawned=1,
 @timeouts={},
 @todo=[],
 @trim_requests=0,
 @waiting=1,
 @workers=[#<Thread:0x007fbc530f1e48@/Users/nownabe/tmp/threadpool/vendor/bundle/ruby/2.2.0/gems/thread-0.1.5/lib/thread/pool.rb:358 sleep>]>
```

**Comment by meh** [comment]
Checking on it, I just realized there are no tests for the pool.

**Comment by meh** [comment]
This should be fixed on 0.1.7, if anyone can test, the specs I wrote locally all pass.

**Comment by nownabe** [comment]
Thank you for fix.
I wrote test for this, please check #37 .
